doc/algorithms/quantum_neural_network.rst

```diff
@@ -0,0 +1,170 @@
+
+Layers in the feedforward neural network above are called **fully connected** as every neuron in a given hidden layer or output layer can be connected to all neurons in the previous layer through the matrix :math:`W`. Over time, specialized versions of layers have been developed to focus on different problems. For example, convolutional layers have a restricted form of connectivity and are suited to machine learning with images. We focus here on fully connected layers as the most general type. Training of neural networks uses variations of the `gradient descent <https://en.wikipedia.org/wiki/Gradient_descent>`_ algorithm on a cost function characterizing the similarity between outputs of the neural network and training data. The gradient of the cost function can be calculated using `automatic differentiation <https://en.wikipedia.org/wiki/Automatic_differentiation>`_, with knowledge of the feedforward network structure.
+
+Quantum neural networks aim to encode neural networks into a quantum system, with the intention of benefitting from quantum information processing. There have been numerous attempts to define a quantum neural network, each with varying advantages and disadvantages. The quantum neural network detailed below, following the work of :cite:`killoran2018continuous`, has a CV architecture and is naturally realized using standard CV gates from Strawberry Fields.
+
+CV implementation
+------------------------------------
+
+A CV quantum neural network layer can be defined as 
+
+.. math:: \mathcal{L} := \Phi \circ \mathcal{D} \circ \mathcal{U}_{2} \circ \mathcal{S} \circ \mathcal{U}_{1},
+
+where :math:`\mathcal{U}_{k}=U_{k}(\boldsymbol{\theta}_{k},\boldsymbol{\phi}_{k})` is an :math:`N` mode interferometer, :math:`\mathcal{D}=\otimes_{i=1}^{N}D(\alpha_{i})` and :math:`\mathcal{S}=\otimes_{i=1}^{N}S(r_{i})` are displacement gates (:class:`~.Dgate`) and squeezing gates (:class:`~.Sgate`) acting on each mode with :math:`\alpha_{i} \in \mathbb{C}` and :math:`r_{i} \in \mathbb{R}`, and finally :math:`\Phi=\otimes_{i=1}^{N}\Phi(\lambda_{i})` is a non-Gaussian gate on each mode with parameter :math:`\lambda_{i} \in \mathbb{R}`. Any non-Gaussian gate such as the cubic phase gate (:class:`~.Vgate`) represents a valid choice, but we recommend the Kerr gate (:class:`~.Kgate`) for simulations in Strawberry Fields as it is diagonal in the Fock basis. The layer is shown below as a circuit:

```

Comment:
Maybe bullet points here again? (Note: I also updated it so that the squeezing parameter is complex)

```suggestion
where

* :math:`\mathcal{U}_{k}=U_{k}(\boldsymbol{\theta}_{k},\boldsymbol{\phi}_{k})` is an :math:`N` mode interferometer, 

* :math:`\mathcal{D}=\otimes_{i=1}^{N}D(\alpha_{i})` is a single mode displacement gate (:class:`~.Dgate`) with complex displacement :math:`\alpha_{i} \in \mathbb{C}`,

 * :math:`\mathcal{S}=\otimes_{i=1}^{N}S(r_{i})` is a single mode squeezing gate (:class:`~.Sgate`) acting on each mode with complex squeezing parameter :math:`z_{i} \in \mathbb{C}`, and
 
 * :math:`\Phi=\otimes_{i=1}^{N}\Phi(\lambda_{i})` is a non-Gaussian gate on each mode with parameter :math:`\lambda_{i} \in \mathbb{R}`.
 
 Any non-Gaussian gate such as the cubic phase gate (:class:`~.Vgate`) represents a valid choice, but we recommend the Kerr gate (:class:`~.Kgate`) for simulations in Strawberry Fields as it is diagonal in the Fock basis. The layer is shown below as a circuit:
```

Reply:
Also, should the squeezing parameter $r$ be a complex squeezing parameter $z$?